Tests/ChessKitTests/BoardTests.swift

```diff
@@ -152,29 +161,54 @@ class BoardTests: XCTestCase {
         // opposite color bishops VS same color bishops
         let fen2 = "k5B1/b7/1b6/8/8/8/8/K7 w - - 0 1"
         let fen3 = "k5B1/1b6/2b5/8/8/8/8/K7 w - - 0 1"
-
+        
         let board2 = Board(position: .init(fen: fen2)!)
         let board3 = Board(position: .init(fen: fen3)!)
-
+        
         XCTAssertFalse(board2.position.hasInsufficientMaterial)
         XCTAssertTrue(board3.position.hasInsufficientMaterial)
-
+        
         // before and after king takes Queen
         let fen4 = "k7/1Q6/8/8/8/8/8/K7 w - - 0 1"
         var board4 = Board(position: .init(fen: fen4)!)
-
+        
         XCTAssertFalse(board4.position.hasInsufficientMaterial)
         board4.move(pieceAt: .a8, to: .b7)
         XCTAssertTrue(board4.position.hasInsufficientMaterial)
     }
     
+    func testThreefoldRepetition() {
+        
+        var board: Board = Board(position: .standard)
+        
+        board.move(pieceAt: .e2, to: .e4)
+        board.move(pieceAt: .e7, to: .e5)
+        
+        board.move(pieceAt: .g1, to: .f3)
+        board.move(pieceAt: .g8, to: .f6)
+        
+        board.move(pieceAt: .f3, to: .g1)
+        board.move(pieceAt: .f6, to: .g8)
+        
+        board.move(pieceAt: .g1, to: .f3)
+        board.move(pieceAt: .g8, to: .f6)
+        
+        board.move(pieceAt: .f3, to: .g1)
+        
+        // check draw before and after the move
+        XCTAssertFalse(board.hashedPositions()[board.position.hashValue] == 3)
+        board.move(pieceAt: .f6, to: .g8)
+        XCTAssertTrue(board.hashedPositions()[board.position.hashValue] == 3)
```

Comment:
Instead of checking `hashedPositions`, I would use a mock `BoardDelegate`. See `testInsufficientMaterial()` and `testFiftyMoveRule()` for examples. Then `hashedPositions()` can be removed and we don't have to expose the implementation details.